internal/provider/providers/porkbun/provider.go

```diff
@@ -148,17 +163,31 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 	return ip, nil
 }
 
-func (p *Provider) deleteALIASRecordIfNeeded(ctx context.Context, client *http.Client) (err error) {
-	aliasRecordIDs, err := p.getRecordIDs(ctx, client, "ALIAS")
+// Deletes the matching record for a specific recordType iff the content matches the expected content value.
+// Returns an error if there is a matching record but an unexpected content value.
+// If there is no matching record, there is nothing to do.
+func (p *Provider) deleteMatchingRecord(
+	ctx context.Context,
+	client *http.Client,
+	recordType string,
+	expectedContent string,
+) error {
+	records, err := p.getRecords(ctx, client, recordType)
 	if err != nil {
-		return fmt.Errorf("getting ALIAS record IDs: %w", err)
-	} else if len(aliasRecordIDs) == 0 {
-		return nil
+		return fmt.Errorf("getting %s records: %w", recordType, err)
```

Comment:
see above comment - getRecords should wrap the record type

```suggestion
		return fmt.Errorf("getting records: %w", err)
```